### PR TITLE
{lang}[iomkl/2018a] R 3.4.4 (+ deps)

### PR DESCRIPTION
**Pull request by edmondac**

### CODE REVIEW

**Comment by edmondac** [comment]
I'm downgrading netCDF to 4.5.0 again, since 4.6.0 gives this build error which I don't have time to fix right now. If you have any ideas then let me know...

```
-- Looking for H5Z_SZIP in hdf5 - found
CMake Error at CMakeLists.txt:707 (MESSAGE):
  HDF5 Requires SZIP, but cannot find libszip or libsz.
```

**Comment by edmondac** [comment]
@boegel  - a question about dependencies... I've been trying to bump everything in this PR to use libxml2-2.9.7. Generally this has been fine as there have been new versions of everything where I've needed to change it in an existing file (e.g. a new GLib-2.54.3-GCCcore-6.4.0.eb that can take the bumped version). However, gettext-0.19.8.1-GCCcore-6.4.0.eb uses libxml2-2.9.4 and there isn't a new version of gettext. What should I do?

**Comment by boegel** [comment]
@edmondac I ran into the same issue, so I added an additional easyconfig for `gettext` that depends on `libxml2` 2.9.7, and added a `versionsuffix`, see #5979 .

I guess it makes sense to make this PR depends on #5979?

Also, do you think it makes sense to put this on hold until R 3.4.4 is released (ETA March 15h)?

I would strongly prefer only having one R version in the `2018a` generation of easyconfigs, and since we're so close to a new R release...

**Comment by edmondac** [comment]
@boegel yes let's wait for R 3.4.4. I'll pick this up again at that point, assuming #5979 has merged by then.

**Comment by edmondac** [comment]
@boegel I had to add Python 3.6.4 back in here, as GEOS and GDAL have it as a dep. Anyway, here's an up-to-date R 3.4.4 built with iomkl that builds and now passes the CI tests...

**Comment by edmondac** [comment]
I had to change the source_urls, as the previous CRAN mirror hadn't picked up R 3.4.4 yet...

File: easybuild/easyconfigs/n/netCDF/netCDF-4.5.0-iomkl-2018a.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name = 'netCDF'
+version = '4.5.0'
```
Comment (boegel) [inline easybuild/easyconfigs/n/netCDF/netCDF-4.5.0-iomkl-2018a.eb]: @edmondac Not used as a dep (anymore), so please remove this easyconfig?

File: easybuild/easyconfigs/n/netCDF/netCDF-4.6.0-iomkl-2018a.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+dependencies = [
+    ('HDF5', '1.10.1'),
+    ('cURL', '7.58.0'),
+    ('Szip', '2.1.1'),
+]
+
+builddependencies = [
+    ('Autotools', '20170619'),
+    ('CMake', '3.10.2'),
+    ('Doxygen', '1.8.13'),
```
Comment (boegel) [inline easybuild/easyconfigs/n/netCDF/netCDF-4.6.0-iomkl-2018a.eb]: bump to 1.8.14?

File: easybuild/easyconfigs/p/PROJ/PROJ-4.9.3-iomkl-2018a.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#
+# Copyright:: Copyright 2014-2015 The Cyprus Institute
+# Authors:: Thekla Loizou <t.loizou@cyi.ac.cy>
+# License:: MIT/GPL
+#
+##
+easyblock = 'ConfigureMake'
+
+name = 'PROJ'
+version = '4.9.3'
```
Comment (boegel) [inline easybuild/easyconfigs/p/PROJ/PROJ-4.9.3-iomkl-2018a.eb]: @edmondac Not used as a dep, so please remove this easyconfig?

File: easybuild/easyconfigs/d/Doxygen/Doxygen-1.8.14-GCCcore-6.4.0.eb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+toolchain = {'name': 'GCCcore', 'version': '6.4.0'}
+
+source_urls = ['http://ftp.stack.nl/pub/users/dimitri/']
+sources = ['%(namelower)s-%(version)s.src.tar.gz']
+checksums = ['d1757e02755ef6f56fd45f1f4398598b920381948d6fcfa58f5ca6aa56f59d4d']
+
+builddependencies = [
+    ('binutils', '2.28'),
+    ('Bison', '3.0.4'),
+    ('CMake', '3.9.1'),
```
Comment (boegel) [inline easybuild/easyconfigs/d/Doxygen/Doxygen-1.8.14-GCCcore-6.4.0.eb]: @edmondac Please use `3.10.2`

File: easybuild/easyconfigs/r/R/R-3.4.4-iomkl-2018a-X11-20180131.eb [hunk 5]
```diff
@@ -0,0 +1,1775 @@
...
+    'source_urls': [
+        'http://cran.r-project.org/src/contrib/Archive/%(name)s',  # package archive
+        'http://cran.r-project.org/src/contrib/',  # current version of packages
+        'http://cran.freestatistics.org/src/contrib',  # mirror alternative for current packages
+    ],
+    'source_tmpl': '%(name)s_%(version)s.tar.gz',
+}
+
+# !! order of packages is important !!
+# packages updated on Dec 4th 2017
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-3.4.4-iomkl-2018a-X11-20180131.eb]: @edmondac Please correct this with the right date?

File: easybuild/easyconfigs/p/Python/Python-3.6.4-iomkl-2018a.eb [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+sources = [SOURCE_TGZ]
+checksums = ['7dc453e1a93c083388eb1a23a256862407f8234a96dc4fae0fc7682020227486']
+
+# python needs bzip2 to build the bz2 package
+dependencies = [
+    ('bzip2', '1.0.6'),
+    ('zlib', '1.2.11'),
+    ('libreadline', '7.0'),
+    ('ncurses', '6.0'),
+    ('SQLite', '3.21.0'),
```
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-3.6.4-iomkl-2018a.eb]: @edmondac Please also include `XZ` as a dependency, see #6065

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 11 out of 12 (12 easyconfigs in this PR)
node2665.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/bcc26534b6015959a88464ab91529957 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 11 out of 12 (12 easyconfigs in this PR)
node2096.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/d7948ace72d93c76b42bdef0b53e0b3b for a full test report.

File: easybuild/easyconfigs/r/R/R-3.4.4-iomkl-2018a-X11-20180131.eb [hunk 7]
```diff
@@ -0,0 +1,1775 @@
...
+        'checksums': ['5aab7671086ef9940e030324651976456f0e84dab35edb7048693ade885228c6'],
+    }),
+    ('ellipse', '0.4.1', {
+        'checksums': ['1a9a9c52195b26c2b4d51ad159ab98aff7aa8ca25fdc6b2198818d1a0adb023d'],
+    }),
+    ('leaps', '3.0', {
+        'checksums': ['55a879cdad5a4c9bc3b5697dd4d364b3a094a49d8facb6692f5ce6af82adf285'],
+    }),
+    ('nloptr', '1.0.4', {
+        'checksums': ['0981a80c46c174a5458bdb8427b2db5428c3daca12ff2c30df7023f143fad7b3'],
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-3.4.4-iomkl-2018a-X11-20180131.eb]: This is wrong somehow, I have `84225b993cb1ef7854edda9629858662cc8592b0d1344baadea4177486ece1eb` for a fresh download of `nloptr_1.0.4.tar.gz`, and also for my cached download which dates from Nov 17th...

Comment (edmondac) [reply easybuild/easyconfigs/r/R/R-3.4.4-iomkl-2018a-X11-20180131.eb]: Yes, very strange. I deleted my tgz, and ran inject-checksums and now it's 84225b...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 11 out of 12 (12 easyconfigs in this PR)
node2544.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/626333eeb6ff78e2e86f0665bd1466c6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 11 out of 12 (12 easyconfigs in this PR)
node2098.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/9bf1dc5da459400dc2948a98d63c76ac for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
node2548.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/1f4a73ed6d8ffbe9334a7cca63c9cd64 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
node2077.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/9f7043a1c016875d3cb6b4f06db3e490 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @edmondac!

**Pull request merged**